src/Common/src/System/Runtime/InteropServices/StringBuffer.cs

```diff
@@ -0,0 +1,385 @@
+            if (sourceIndex < 0 || sourceIndex > source.Length) throw new ArgumentOutOfRangeException("sourceIndex");
+            if (count < 0) count = source.Length - sourceIndex;
+            if (source.Length < sourceIndex + count) throw new ArgumentOutOfRangeException("count");
+
+            if (count == 0) return;
+            ulong lastIndex = bufferIndex + (ulong)count;
+            if (_length < lastIndex) Length = lastIndex;
+
+            fixed (char* content = source)
+            {
+                Buffer.MemoryCopy(
+                    source: content + sourceIndex,
+                    destination: CharPointer + bufferIndex,
+                    destinationSizeInBytes: checked((long)ByteCapacity),
+                    sourceBytesToCopy: count * sizeof(char));
+            }
+        }
+
+        /// <summary>
+        /// Split the contents into strings via the given split characters.
+        /// </summary>
+        /// <exception cref="OverflowException">Thrown if the substring is too big to fit in a string.</exception>
+        unsafe public IEnumerable<string> Split(char splitCharacter)
+        {
+            var strings = new List<string>();
+            char* start = CharPointer;
+            char* current = start;
+
+            ulong length = _length;
+
+            for (ulong i = 0; i < length; i++)
+            {
+                if (splitCharacter == *current)
+                {
+                    // Split
+                    strings.Add(new string(value: start, startIndex: 0, length: checked((int)(current - start))));
+                    start = current + 1;
+                }
+
+                current++;
+            }
+
+            strings.Add(new string(value: start, startIndex: 0, length: checked((int)(current - start))));
+
+            return strings;
+        }
+
+        /// <summary>
+        /// True if the buffer contains the given character.
+        /// </summary>
+        public unsafe bool Contains(char value)
+        {
+            char* start = CharPointer;
+            ulong length = _length;
+
+            for (ulong i = 0; i < length; i++)
+            {
+                if (*start++ == value) return true;
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// True if the buffer contains any of the specified characters.
+        /// </summary>
+        public unsafe bool Contains(params char[] values)
+        {
+            if (values == null || values.Length == 0) return false;
+
+            char* start = CharPointer;
+            ulong length = _length;
+
+            for (ulong i = 0; i < length; i++)
+            {
+                if (ContainsChar(values, *start)) return true;
+                start++;
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Trim the specified values from the end of the buffer.
+        /// </summary>
+        public unsafe void TrimEnd(params char[] values)
```

Comment:
Let's remove the params.  Since this type is all about perf, folks should know that they're incurring an allocation if they pass in characters not already in an array.  Additionally, string.TrimEnd has different behavior from what's coded here if no chars are provided, and I think it'd be confusing for TrimEnd() to be a nop rather than trimming whitespace from the end.  As with the other helpers, if this isn't used, let's remove it entirely. (I'll stop commenting on that aspect of the helpers.)